Sources/MapboxNavigation/NavigationMapView+BuildingHighlighting.swift

```diff
@@ -15,13 +15,99 @@ extension NavigationMapView {
      - returns: Bool indicating if number of buildings found equals number of coordinates supplied.
      */
     @discardableResult public func highlightBuildings(at coordinates: [CLLocationCoordinate2D], in3D extrudesBuildings: Bool = true) -> Bool {
-        return false
+        let _ = coordinates.map({buildingIdentifier(at: $0) })
+        DispatchQueue.main.asyncAfter(deadline: .now() + handleAsyncTime, execute: {self.addBuildingsLayer(with: self.foundBuildingIds, in3D: extrudesBuildings)})
+        return foundBuildingIds.count == coordinates.count
     }
     
     /**
      Removes the highlight from all buildings highlighted by `highlightBuildings(at:in3D:)`.
      */
     public func unhighlightBuildings() {
-        
+        guard let _ = try? mapView.style.getLayer(with: IdentifierString.buildingExtrusionLayer, type: FillExtrusionLayer.self).get() else { return }
+        foundBuildingIds = Set<Double>()
+        let _ = mapView.style.removeStyleLayer(forLayerId: IdentifierString.buildingExtrusionLayer)
     }
+
+    private func featureIdentifierToDouble(features: [Feature]) -> Double? {
+        if let featureIdentifier = features.first?.identifier {
```

Comment:
Early return with `guard let` would be better here.

Reply:
Thanks for the suggestion! I'll work on that.

Reply:
Thanks for the suggestion! I'll work on that.